pkg/agent/core/ngt/service/ngt_stateful_test.go

```diff
@@ -514,9 +515,13 @@ var (
 			state commands.State,
 			result commands.Result,
 		) *gopter.PropResult {
+			st := state.(*ngtState)
 			rc := result.(*resultContainer)
-
-			if rc.err != nil {
+			if rc.err != nil &&
+				!(st.states[idA] != INDEXED &&
+					st.states[idB] != INDEXED &&
+					st.states[idC] != INDEXED &&
+					errors.Is(rc.err, errors.ErrEmptySearchResult)) {
```

Comment:
I think it should be like the following.
```go
			if rc.err != nil {
				if st.states[idA] != INDEXED &&
					st.states[idB] != INDEXED &&
					st.states[idC] != INDEXED {
					if !errors.Is(rc.err, errors.ErrEmptySearchResult) {
						return &gopter.PropResult{
							Status: gopter.PropFalse,
							Error:  rc.err,
							Labels: []string{
								"Search",
								"there's no index but it doesn't return ErrEmptySearchResult",
								rc.err.Error(),
							},
						}
					}
				} else {
					return &gopter.PropResult{
						Status: gopter.PropFalse,
						Error:  rc.err,
						Labels: []string{
							"Search",
							"error",
							rc.err.Error(),
						},
					}
				}
			}
```
It is easier to find what is problem.